File: app/src/main/java/com/water/broadcasesample/AsyncTaskActivity.java
package com.water.broadcasesample;

import android.os.AsyncTask;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ProgressBar;
import android.widget.TextView;

public class AsyncTaskActivity extends AppCompatActivity {
    private Button btnDownload;
    private ProgressBar pbSchedule;
    private TextView tvTitle;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_async_task);
        pbSchedule= findViewById(R.id.pbSchedule);
        tvTitle= findViewById(R.id.tvTitle);

        btnDownload = findViewById(R.id.btnDownload);
        btnDownload.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                DownloadTask dt = new DownloadTask();
                dt.execute(100);
            }
        });
    }

    private class DownloadTask extends AsyncTask<Integer, Integer, String> {

        @Override
        protected void onPreExecute() {
            //第一个执行方法
            super.onPreExecute();
        }

        @Override
        protected String doInBackground(Integer... params) {
            //第二个执行方法,onPreExecute()执行完后执行
            for(int i=0;i<=100;i++){
                publishProgress(i);

                try {
                    Thread.sleep(params[0]);
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }
            }
            return "执行完毕";
        }

        @Override
        protected void onProgressUpdate(Integer... progress) {
            pbSchedule.setProgress(progress[0]);
            tvTitle.setText(progress[0]+"%");
            super.onProgressUpdate(progress);
        }

        @Override
        protected void onPostExecute(String result) {
            setTitle(result);
            super.onPostExecute(result);
        }
    }
}

File: app/src/main/kotlin/com/github/androidhappyclub/broadcasesample/AsyncTaskActivity.kt
package com.github.androidhappyclub.broadcasesample

import android.os.AsyncTask
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.ave.vastgui.tools.view.extension.refreshWithInvalidate
import com.ave.vastgui.tools.viewbinding.viewBinding
import com.github.androidhappyclub.broadcasesample.databinding.ActivityAsyncTaskBinding
import com.github.androidhappyclub.broadcasesample.log.mLogFactory

class AsyncTaskActivity : AppCompatActivity(R.layout.activity_async_task) {

    private val binding by viewBinding(ActivityAsyncTaskBinding::bind)
    private val logger = mLogFactory.getLog(AsyncTaskActivity::class.java)

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding.btnDownload.setOnClickListener { v: View? ->
            val dt = DownloadTask()
            dt.execute(100)
        }

    }

    private inner class DownloadTask : AsyncTask<Int, Int, String>() {

        override fun onPreExecute() {
            // 第一个执行方法
            super.onPreExecute()
            logger.i("下载任务开始")
        }

        override fun doInBackground(vararg params: Int?): String {
            // 第二个执行方法,onPreExecute()执行完后执行
            for (i in 0..100) {
                publishProgress(i)
                try {
                    Thread.sleep((params[0] ?: 0).toLong())
                } catch (e: InterruptedException) {
                    e.printStackTrace()
                }
            }
            return "执行完毕"
        }

        override fun onProgressUpdate(vararg progress: Int?) {
            binding.pbSchedule.refreshWithInvalidate {
                mCurrentProgress = progress[0]?.toFloat() ?: 0f
            }
            binding.tvTitle.text = "${progress[0]}%"
            super.onProgressUpdate(*progress)
        }

        override fun onPostExecute(result: String) {
            title = result
            super.onPostExecute(result)
        }
    }
}